### PR TITLE
fix: fixing sync return format missing flag layer, adding full e2e suite

### PR DESCRIPTION
**Pull request by aepfli**
with https://github.com/open-feature/flagd/pull/1797 we introduced this bug, that the format of the response is not correct.

current state:
```javacript
{ 
    flagConfiguration: {/* flag object */}
}
```
should be:
```javacript
{ 
    flagConfiguration: {
       flags: {/* flag object */}
    }
}
```

Clarification from @toddbaert - this is an UNRELEASED bug so far.

### CODE REVIEW

**Comment by aepfli** [comment]
we definitely need to get everything going for https://github.com/open-feature/flagd/pull/1821 to prevent this issue in the future

**Comment by toddbaert** [comment]
Yep, :100: I found this late last week and was intending to open a bug as well. We need to enhance our testing to run in-process tests to find this - there's much less surface area to test with in-process mode on flagd itself (since evaluation happens in the workload) but it's required to find things like this.

Thanks!

File: test/integration/integration_test.go [hunk 1]
```diff
@@ -1,13 +1,183 @@
...
+	})
+	defer runner.Cleanup()
+
+	// Define feature paths - using flagd-testbed gherkin files
+	featurePaths := []string{
+		"./",
+	}
+
+	// Run tests with InProcess-specific tags
+	tags := "@in-process && ~@unixsocket&& ~@metadata && ~@contextEnrichment && ~@customCert && ~@forbidden && ~@sync-port && ~@sync-payload && ~@grace && ~@targetURI"
```
Comment (aepfli) [inline test/integration/integration_test.go]: i am not sure why targetURI is failing here, in the provider it is working fine, did we maybe add another regression somehow?

Comment (aepfli) [reply test/integration/integration_test.go]: the test runs fine with this changes as e2e test in the flagd-golang-provider. so i assume that this might be some kind of strange execution issue. but not worth further exploration

Comment (aepfli) [reply test/integration/integration_test.go]: okay updated the go moduels and now it seems to work as intended

File: test/integration/integration_test.go
Comment (alexandraoberaigner) [inline test/integration/integration_test.go:1]: [Suggestion] maybe we could use a similar naming/file structure as in the go flagd provider to make it easier to find things? eg. have put rpc, file, and inprocess test functions in separate files. wdyt?

File: test/integration/go.mod
Comment (alexandraoberaigner) [inline test/integration/go.mod:1]: [Q] are these new imports all required for the testcontainers? 👀

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -124,32 +124,5 @@ jobs:
...
-          -f file:${{ github.workspace }}/test-harness/flags/custom-ops.json \
-          -f file:${{ github.workspace }}/test-harness/flags/evaluator-refs.json \
-          -f file:${{ github.workspace }}/test-harness/flags/zero-flags.json \
-          -f file:${{ github.workspace }}/test-harness/flags/edge-case-flags.json &
-
-      - name: Run envoy proxy in background
-        run: |
-          envoy -c ./test/integration/config/envoy.yaml &
-
       - name: Run evaluation test suite
```
Comment (alexandraoberaigner) [inline .github/workflows/build.yaml]:
```suggestion
      - name: Run test suite
```
[Q] Its not only the evaluation test suite anymore, is it? Should we rename this? Open for other names too

**Review by toddbaert — APPROVED** [review]
Thanks for the fix, but thanks way more for integrating the entire e2e suite. Very very good to have.

**Comment by toddbaert** [comment]
@aepfli ... @alexandraoberaigner has some small questions/nits, but not blockers for me. I've rebased, so I'll merge this _soon_ unless you want to address those small nits.
